web/src/components/ThemeContext.tsx

```diff
@@ -30,12 +31,18 @@ type ThemeContainerProps = {
 }
 
 export const ThemeContainer = ({ children, contentDirection }: ThemeContainerProps): ReactElement => {
-  const [themeType, setThemeType] = useState<ThemeType>(() => getSystemTheme())
+  const { value: themeType, updateLocalStorageItem: setThemeType } = useLocalStorage<ThemeType>({
+    key: 'themeType',
+    initialValue: getSystemTheme(),
+  })
 
   useEffect(() => {
     const handleSystemChange = () => {
-      const systemTheme = getSystemTheme()
-      setThemeType(prev => (prev !== systemTheme ? systemTheme : prev))
+      const storedTheme = localStorage.getItem('themeType')
+      if (!storedTheme || storedTheme === 'null') {
+        const currentTheme = getSystemTheme()
+        setThemeType(currentTheme)
+      }
```

Comment:
🔧 This is never executed as `useLocalStorage` directly persists the value to the local storage, so it is always set when you check for `storedTheme`. Removing this whole `useEffect` therefore still has the same behavior.
If you want to automatically update the theme if the browser settings change, just remove the `if`-clause.

Reply:
I refactored it so the toggle state is saved as well

Reply:
My message from above is still true. This will never be executed and can therefore be removed. 🤔

Reply:
no, this is needed to keep the state after refresh, without it, it will always go back to the system theme

Reply:
I cannot reproduce this. For me this works just fine after deleting lines L43 to L58 :/ Can you please test again?

Reply:
you are right, I think I missed something when I was testing it :see_no_evil: